src/common/interfaces/IGatewayHandlers.sol

```diff
@@ -158,102 +158,41 @@ interface IRequestManagerGatewayHandler {
     // --- Deposits ---
     /// @notice Fulfills pending deposit requests after successful epoch execution on Hub.
     ///         The amount of shares that can be claimed by the user is minted and moved to the escrow contract.
-    ///         The MaxMint bookkeeping value is updated.
+    ///         The maxMint and claimableCancelDepositRequest bookkeeping values are updated.
     ///         The request fulfillment can be partial.
     /// @dev    The shares in the escrow are reserved for the user and are transferred to the user on deposit
     ///         and mint calls.
+    /// @dev    The cancelled and fulfilled amounts are both non-zero iff the cancellation was queued.
+    ///         Otherwise, either of the two must always be zero.
     function fulfillDepositRequest(
         PoolId poolId,
         ShareClassId scId,
         address user,
         AssetId assetId,
-        uint128 assets,
-        uint128 shares
-    ) external;
-
-    /// @notice Fulfills deposit request cancellation after successful epoch execution on Hub.
-    ///         The amount of assets that can be claimed by the user is locked in the escrow contract.
-    ///         Updates claimableCancelDepositRequest bookkeeping value. The cancellation order execution can be
-    ///         partial.
-    /// @dev    The assets in the escrow are reserved for the user and are transferred to the user during
-    ///         claimCancelDepositRequest calls.
-    ///         `fulfillment` represents the decrease in `pendingDepositRequest`.
-    ///         This is a separate parameter from `assets` since there can be some precision loss when calculating this,
```

Comment:
Artifact of v2 Centrifuge Chain implementation that luckily doesn't hold true for v3 (see [Slack thread](https://kflabs.slack.com/archives/C07PG2EUR9C/p1736937649220279?thread_ts=1736933248.761209&cid=C07PG2EUR9C)). With v3, we have always returned the exact same value here because we "swap" only during claiming at the price which was set during the approval of the corresponding epoch. So `fulfillment` and `assets` are always the same.

Reply:
Really nice improvement actually!